modules/repo_sync/src/Service/Client.php

```diff
@@ -194,14 +197,14 @@ protected function withCurl(string $method, string $url, array $headers, ?callab
       $callback($ch);
     }
 
-    if ($rawresult = curl_exec($ch)) {
+    if (!($rawresult = curl_exec($ch))) {
+      throw new DevportalRepoSyncConnectionException('Curl handler error.');
```

Comment:
This could be more verbose. Maybe use this: http://php.net/manual/en/function.curl-error.php